### ISSUE TITLE
NPE in the Authenticator when executing a query

### ISSUE THREAD

**Issue by SirStoke** [orientechnologies/orientdb#9809]
### OS: Linux (container)

We encountered a NPE in the Authenticator class while executing a simple query from a java remote client. The query was a very simple `SELECT/WHERE`, but it was executed under slight concurrency

## Expected behavior

Query is executed correctly

## Actual behavior  
The client failed with `orientechnologies.orient.core.exception.OSecurityAccessException: Invalid authentication info for access to the database com.orientechnologies.orient.core.metadata.security.auth.OTokenAuthInfo`

And the instance logged the following exception:

```
java.lang.NullPointerException
	at com.orientechnologies.orient.core.security.authenticator.ODatabaseUserAuthenticator.authenticate(ODatabaseUserAuthenticator.java:26)
	at com.orientechnologies.orient.core.security.ODefaultSecuritySystem.authenticate(ODefaultSecuritySystem.java:218)
	at com.orientechnologies.orient.core.metadata.security.OSecurityShared.securityAuthenticate(OSecurityShared.java:243)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.internalOpen(ODatabaseDocumentEmbedded.java:207)
	at com.orientechnologies.orient.core.db.OrientDBEmbedded.open(OrientDBEmbedded.java:507)
	at com.orientechnologies.orient.server.OServer.openDatabase(OServer.java:830)
	at com.orientechnologies.orient.server.OClientConnection.init(OClientConnection.java:246)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.onBeforeOperationalRequest(ONetworkProtocolBinary.java:543)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.sessionRequest(ONetworkProtocolBinary.java:302)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.execute(ONetworkProtocolBinary.java:239)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:68)
```

## Steps to reproduce

Sadly we aren't able to reproduce it reliably

**Comment by johny65** [comment]
We have the same issue but it occurs randomly in certain queries, retrying the query is executed fine so we are for now ignoring it. We are using version 3.2.0.

**Comment by tglman** [comment]
Hi, 
I did a small refactor that should improve the current status, it landed in 3.2.7 just released, could you double check it?

Regards

**Comment by tglman** [comment]
Closing this, please re-open if you can still reproduce it
